LLMNeedleHaystackTester.py

```diff
@@ -62,93 +63,72 @@ def __init__(self,
+                    self.document_depth_percents = [self.logistic(x) for x in
+                                                    np.linspace(document_depth_percent_min, document_depth_percent_max,
+                                                                document_depth_percent_intervals)]
         else:
             self.document_depth_percents = document_depth_percents
 
         if document_depth_percent_interval_type not in [None, "linear", "sigmoid"]:
-            raise ValueError("document_depth_percent_interval_type must be either None, 'linear' or 'sigmoid'. If you'd like your own distribution give a list of ints in via document_depth_percent_intervals")
-        
-        if model_provider not in ["OpenAI", "Anthropic"]:
-            raise ValueError("model_provider must be either 'OpenAI' or 'Anthropic'")
-        
-        if model_provider == "Anthropic" and "claude" not in model_name:
-            raise ValueError("If the model provider is 'Anthropic', the model name must include 'claude'. See https://docs.anthropic.com/claude/reference/selecting-a-model for more details on Anthropic models")
-        
-        self.openai_api_key = openai_api_key or os.getenv('OPENAI_API_KEY')
-        self.model_name = model_name
-
-        if not self.openai_api_key and not os.getenv('OPENAI_API_KEY'):
-            raise ValueError("Either openai_api_key must be supplied with init, or OPENAI_API_KEY must be in env. Used for evaluation model")
-        else:
-            self.openai_api_key = openai_api_key or os.getenv('OPENAI_API_KEY')
-        
-        self.anthropic_api_key = anthropic_api_key or os.getenv('ANTHROPIC_API_KEY')
+            raise ValueError(
+                "document_depth_percent_interval_type must be either None, 'linear' or 'sigmoid'. If you'd like your own distribution give a list of ints in via document_depth_percent_intervals")
 
-        if self.model_provider == "Anthropic":
-            if not self.anthropic_api_key and not os.getenv('ANTHROPIC_API_KEY'):
-                raise ValueError("Either anthropic_api_key must be supplied with init, or ANTHROPIC_API_KEY must be in env.")
-            else:
-                self.anthropic_api_key = anthropic_api_key or os.getenv('ANTHROPIC_API_KEY')
-            
-        if not self.model_name:
-            raise ValueError("model_name must be provided.")
-        
-        if model_provider == "OpenAI":
-            self.model_to_test = AsyncOpenAI(api_key=self.openai_api_key)
-            self.enc = tiktoken.encoding_for_model(self.model_name)
-        elif model_provider == "Anthropic":
-            self.model_to_test = AsyncAnthropic(api_key=self.anthropic_api_key)
-            self.enc = Anthropic().get_tokenizer()
-        
-        self.model_to_test_description = model_name
-        self.evaluation_model = ChatOpenAI(model="gpt-4", temperature=0, openai_api_key = self.openai_api_key)
+
+        if evaluation_method == 'gpt4':
```

Comment:
I see `'gpt4'` very often, maybe we can consider having variable for example `OpenAIEvaluator.MODEL_NAME = 'gpt4'`.  
This way we may easily add support for more models like `GPT-4 Turbo` by subclassing `OpenAIEvaluator` and setting few variables.

Reply:
Thank you for the feedback on the model specification. You're correct that the "Needle in Haystack" tester is designed to be versatile and accepts a range of models as runtime parameters. Users have the flexibility to test with various models such as GPT-3.5, GPT-4, or GPT-4 Turbo, catering to diverse testing needs without any hardcoded constraints.

Regarding the model response evaluation, we currently support two methods:

substring_match: A straightforward and effective method for simpler cases.
Utilizing GPT-4: Leveraging GPT-4's advanced capabilities for more complex evaluations.
I concur with the suggestion to avoid hardcoding the model name in the code. Moving towards defining the model as a configurable constant would indeed enhance code modularity and maintainability. This change will also align with our aim for greater flexibility and adaptability in the testing framework

Created issue #10 to track this enhancement.